src/Dockerfile

```diff
@@ -83,11 +83,12 @@ RUN if   [ "$TARGETPLATFORM" = "linux/amd64" ];    then FTLARCH=amd64; \
     && echo "Arch: ${TARGETPLATFORM}, FTLARCH: ${FTLARCH}" \
     && curl -sSL "https://ftl.pi-hole.net/${FTL_BRANCH}/pihole-FTL-${FTLARCH}" -o /usr/bin/pihole-FTL \
     && chmod +x /usr/bin/pihole-FTL \
-    && readelf -h /usr/bin/pihole-FTL || cat /usr/bin/pihole-FTL
+    && readelf -h /usr/bin/pihole-FTL || (cat /usr/bin/pihole-FTL && echo "Error with downloaded FTL binary" && exit 1)
```

Comment:
Looking at this, I'm not actually sure if `cat` is such a wise idea here. Maybe the download was just a bit corrupted - doors of hell may open for you when you `cat` 20 megabytes of raw binary data into the `docker` log

Reply:
Oh... I was writing another suggestion while you was talking here.

Reply:
Also: why did `curl` actually succeed above for an error 404?...

Reply:
Interesting, will investigate

Reply:
`curl` still succeeds because it was a valid web request - it retrieved the 404 page. 

Now, I could add in some additional checks here to ensure the code is `200` - but seems like additional complexity (and negative readability) for something that has the same net effect in the build process.